File: src/main/java/com/partymakers/shareparty/domain/party/port/PartyRoomRepository.java
package com.partymakers.shareparty.domain.party.port;

import com.partymakers.shareparty.domain.party.entity.PartyRoom;

import java.util.Optional;

public interface PartyRoomRepository {
    PartyRoom save(PartyRoom partyRoom);
    Optional<PartyRoom> findById(long id);
    Iterable<PartyRoom> findAll();
}

File: src/main/kotlin/com/partymakers/shareparty/domain/party/port/PartyRoomRepository.kt
package com.partymakers.shareparty.domain.party.port

import com.partymakers.shareparty.domain.party.entity.PartyRoom
import java.util.Optional

interface PartyRoomRepository {
    fun save(partyRoom: PartyRoom): PartyRoom
    fun findById(id: Long): Optional<PartyRoom>
    fun findAll(): Iterable<PartyRoom>
} 